sdk/keyvault/keyvault-keys/src/index.ts

```diff
@@ -479,34 +502,46 @@ export class KeysClient {
   /**
    * The delete operation applies to any key stored in Azure Key Vault. Individual versions
    * of a key can not be deleted, only all versions of a given key at once.
+   *
+   * This function returns a Long Running Operation poller that allows you to wait indifinetly until the key is deleted.
+   *
    * This operation requires the keys/delete permission.
    *
    * Example usage:
    * ```ts
-   * let client = new KeysClient(url, credentials);
-   * let result = await client.deleteKey("MyKey");
+   * const client = new KeysClient(url, credentials);
+   * const poller = await client.beginDeleteKey("MyKey");
+   *
+   * // Serializing the poller
+   * const serialized = poller.toJSON();
+   * // A new poller can be created with:
+   * // await client.beginDeleteKey("MyKey", { resumeFrom: serialized });
+   *
+   * // Waiting until it's done
+   * const deletedKey = await poller.pollUntilDone();
+   * console.log(deletedKey);
    * ```
    * @summary Deletes a key from a specified key vault.
-   * @param vaultBaseUrl The vault name, for example https://myvault.vault.azure.net.
    * @param name The name of the key.
    * @param [options] The optional parameters
    */
-  public async deleteKey(name: string, options?: RequestOptions): Promise<DeletedKey> {
-    const requestOptions = (options && options.requestOptions) || {};
-    const span = this.createSpan("deleteKey", requestOptions);
+  public beginDeleteKey(name: string, options?: BeginDeleteKeyOptions): DeleteKeyPoller {
+    // Making a fake client to reduce the duplicated code
+    const pollerClient: KeyClientInterface = {
+      deleteKey: this.deleteKey.bind(this),
+      getDeletedKey: this.getDeletedKey.bind(this)
+    };
 
-    let response: DeleteKeyResponse;
-    try {
-      response = await this.client.deleteKey(
-        this.vaultBaseUrl,
-        name,
-        this.setParentSpan(span, requestOptions)
-      );
-    } finally {
-      span.end();
-    }
+    const poller = new DeleteKeyPoller({
+      name,
+      client: pollerClient,
+      ...options
+    });
 
-    return this.getKeyFromKeyBundle(response);
+    // This will initialize the poller's operation (the deletion of the key).
+    poller.poll();
```

Comment:
this should await, right?

Reply:
Nope, I'm trying to avoid double awaits. Since there's no value in the intermediate values of this poller, I'm thinking we can just return the poller and have people call just one await when they `await client.beginDeleteKey().pollUntilDone()`, instead of `await (await client.beginDeleteKey()).pollUntilDone()`.

Reply:
I'll change it

Reply:
Yes, so that any errors that prevent this request from being submitted are raised at this point.

Reply:
yes we should be awaiting this. we want the first request to have completed before this method returns.